src/EVM/SMT.hs

```diff
@@ -362,138 +391,85 @@ declareBufs names = SMT2 ("; buffers" : fmap declareBuf names <> ("; buffer leng
   where
     declareBuf n = "(declare-const " <> n <> " (Array (_ BitVec 256) (_ BitVec 8)))"
     declareLength n = "(define-const " <> n <> "_length" <> " (_ BitVec 256) (bufLength " <> n <> "))"
-    cexvars = CexVars
-      { calldataV = mempty
-      , buffersV = (fmap toLazyText names)
-      , blockContextV = mempty
-      , txContextV = mempty
-      }
+    cexvars = mempty{buffersV = fmap toLazyText names}
+
+
+referencedBufsGo :: Expr a -> [Builder]
+referencedBufsGo = \case
+  AbstractBuf s -> [fromText s]
+  _ -> []
 
 referencedBufs :: Expr a -> [Builder]
-referencedBufs expr = nubOrd (foldExpr go [] expr)
-  where
-    go :: Expr a -> [Builder]
-    go = \case
-      AbstractBuf s -> [fromText s]
-      _ -> []
+referencedBufs expr = nubOrd $ foldExpr referencedBufsGo [] expr
 
 referencedBufs' :: Prop -> [Builder]
-referencedBufs' = \case
```

Comment:
Rewrote a bunch of such functions using `foldProp` and hoisted the `go` helper outside of their `Expr` counterpart to use in both of them.